src/WebJobs.Script/Description/CSharp/CSharpFunctionInvoker.cs

```diff
@@ -175,9 +183,17 @@ private void RestorePackages()
             catch (Exception ex)
             {
                 TraceWriter.Error(ex.Message, ex is CompilationErrorException ? null : ex);
+
+                startedEvent.Success = false;
+                TraceWriter.Error(ex.Message, ex);
+
                 TraceWriter.Verbose("Function completed (Failure)");
                 throw;
             }
+            finally
+            {
+                _metrics.EndEvent(startedEvent);
+            }
         }
```

Comment:
I thought about that as well and think it is a good idea. The reason I didn't is because there are some slight differences in error handling/reporting across the invokers. I think I'll leave this in the bucket of additional refactoring of commonalities that I've been doing over time across the invokers.